appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -441,6 +441,40 @@ Blockly.ReplMgr.putYail = (function() {
             poll();
 
         },
+        'chunker' : (function() {
+            var seq = 0;
+            var gensym = function() {
+                seq += 1;
+                return 'Q' + seq;
+            };
+
+            var chunker = function(input) {
+                var length = input.length;
+                var chunklen = 8192;
+                if (length < chunklen) {
+                    return input;
+                }
+                var chunks = [];
+                while (length > 0) {
+                    var clen = Math.min(length, chunklen);
+                    chunks.push(input.substring(0, clen));
+                    input = input.substring(clen);
+                    length = input.length;
+                }
+                var symbol = gensym();
+                var retval = [];
+                retval.push('(define ' + symbol + ' "")');
+                chunks.forEach(function(item, index) {
+                    item = item.replace(/\\/g, '\\\\');
```

Comment:
Can we refactor it such that the substitution occurs once before we chunk? This would allow us to have a larger chunk size close to 16k to reduce the number of messages (since we wouldn't need the extra buffer since the code is pre-expanded). At the moment, we cut into potentially 2x the number of needed messages and then expand, potentially leaving a lot of extra room per message (really though it's probably limited to modulo the network frame size). The edge case would be if we cut where a `\` is at the end of the string (but not preceded by another `\`).